### DISCUSSION TITLE
Sharding to be or not to be?

### DISCUSSION THREAD

**Discussion by supercaligit** [ravendb/ravendb#18173] [Q&A]
The documentation recommends that sharding not be used for database size less that 100Gb, because clearly the overhead outweighs the benefits. So if I start with a nonsharded database and then expect to migrate to a sharded database I have the following questions. 
- In a non sharded database , should one adopt the suffix method to anchor related documents to same bucket so that when you move to sharded database those documents would stay together??
- If you migrate from a nonsharded to a sharded database, are related documents automatically maintained on same shard?

**Comment by ayende** [comment]
* If you use the suffix method, when you move to sharded, we'll keep them in the same bucket, yes.
* Are you *expecting* to grow to > 100GB in the next year? Or is this planning for the future?
